tox.ini

```
@@ -1,21 +1,28 @@
 [tox]
-envlist=py38
-skipsdist=True
+envlist = py38
+skipsdist = True
 
 [testenv]
 # one virtual env for all test - without this line tox will install .tox/all env to run the 'all' suit
-envdir = {toxinidir}/.tox
+envdir = {toxworkdir}/common
 
 deps =
     -rrequirements.txt
 
 commands =
-    pytest -v -m "not slow" --strict-markers -W ignore::DeprecationWarning {posargs}
+    pytest -m "not slow" {posargs}
 
-[testenv:verbose]
+[testenv:all]
 commands =
-    pytest -sxv {posargs}
+    pytest --cov=. --cov-report term-missing {posargs}
 
-[testenv:all]
+# runs on github workflow
+[testenv:ci]
 commands =
-    pytest -v --strict-markers -W ignore::DeprecationWarning --cov=. --cov-report term-missing {posargs}
+    pytest --cov=. --cov-report term --cov-report=xml
+
+[pytest]
+addopts = -v -W once::DeprecationWarning --strict-markers
+ignore = .tox .git data
```

adding .tox and .git to the ignore list is fine, but I'd prefer not to add data. There may not be any tests in there now, but I could see adding data sanity tests later and silently ignoring them wouldn't be good.